controllers/capabilities/tenant_controller.go

```diff
@@ -95,119 +107,116 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		reqLogger.V(1).Info(string(jsonData))
 	}
 
-	masterAccessToken, err := r.fetchMasterCredentials(tenantR)
-	if err != nil {
-		reqLogger.Error(err, "Error fetching master credentials secret")
-		// Error reading the object - requeue the request.
-		return ctrl.Result{}, err
+	// reconcileErr can be an error of different types - generic k8s err, isInvalid, waitError, isOrphaned
+	// different errors are handled differently when it comes to requeueing the reconciliation loop
+	statusReconciler, reconcileErr := r.specReconciler(tenantR, reqLogger)
+
+	// statusUpdateError can be an error when making the update call to the resource
+	res, statusUpdateErr := statusReconciler.Reconcile()
+	if statusUpdateErr != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to update tenantCR: %w", statusUpdateErr)
 	}
 
-	insecureSkipVerify := controllerhelper.GetInsecureSkipVerifyAnnotation(tenantR.GetAnnotations())
-	portaClient, err := controllerhelper.PortaClientFromURLString(tenantR.Spec.SystemMasterUrl, masterAccessToken, insecureSkipVerify)
-	if err != nil {
-		reqLogger.Error(err, "Error creating porta client object")
-		// Error reading the object - requeue the request.
-		return ctrl.Result{}, err
+	// Requeue if either object or status were updated
+	if res.Requeue {
+		return reconcile.Result{}, nil
```

Comment:
This looks off a bit so suggestions are very much welcomed.
The reasoning behind sending empty result with nil (which equals to no requeue) is that the only time we can land here is when status or spec got updated, which under the hood, automatically triggers a reconcile. 

For any error's we handle the requeue or not (based on error types).